horovod/runner/elastic/discovery.py

```diff
@@ -39,13 +52,48 @@ def get_event(self):
     def set_event(self):
         self._event.set()
 
+    def _in_cooldown_period(self, current_time):
+        return self._cooldown_period_end_ts > current_time
+
+
+    def _set_cooldown_period(self, current_time):
+        if self._cooldown_lower_limit == -1 or self._cooldown_upper_limit == -1:
+            return
+        self._blacklist_count += 1
+        def _exponential_backoff_time():
```

Comment:
please add an empty line before and after this def

Reply:
Added